src/hotspot/cpu/s390/assembler_s390.inline.hpp

```diff
@@ -1364,6 +1405,27 @@ inline unsigned int Assembler::instr_len(unsigned char *instr) {
   }
 }
 
+// Instruction must be stored right-justified in argument.
+inline unsigned int Assembler::instr_len(unsigned long instr) {
+  int len_bits = instr >> (48-2);
+  if (len_bits == 0) len_bits = instr >> (32-2);
+  if (len_bits == 0) len_bits = instr >> (16-2);
```

Comment:
Tricky, but looks correct.

Reply:
Want some detailed comment here?

Reply:
I can understand it, but a short comment would be appreciated. E.g. length can only be >2 if first 2 bits are != 0.

Reply:
Will rework with next commit.